packages/server/lib/controllers/oauth.controller.ts

```diff
@@ -479,12 +481,27 @@ class OAuthController {
                 return publisher.notifyErr(res, channel, providerConfigKey, connectionId, error);
             }
 
+            if (template.authorization_params && missesInterpolationParamInObject(template.authorization_params, connectionConfig)) {
+                const error = WSErrBuilder.InvalidConnectionConfig('authorization_params', JSON.stringify(connectionConfig));
+                await logCtx.error(error.message, { connectionConfig });
+                await logCtx.failed();
+
+                return publisher.notifyErr(res, channel, providerConfigKey, connectionId, error);
+            }
+
+            if (template.token_params && missesInterpolationParamInObject(template.token_params, connectionConfig)) {
+                const error = WSErrBuilder.InvalidConnectionConfig('token_params', JSON.stringify(connectionConfig));
+                await logCtx.error(error.message, { connectionConfig });
+                await logCtx.failed();
+
+                return publisher.notifyErr(res, channel, providerConfigKey, connectionId, error);
+            }
```

Comment:
Seems like a waste of resources to do it multiple times, not for this PR but we should just do everything and check the full object recursively at the end

Reply:
I agree that a recursive check for the entire object at once would be more efficient and reduce redundancy.